test/jbuilder_template_test.rb

```diff
@@ -123,23 +123,39 @@ def assert_collection_rendered(result, context = nil)
 
   test "partial! renders partial" do
     result = jbuild(<<-JBUILDER)
-      json.partial! "partial"
+      json.partial! "partial", foo: 'hello'
     JBUILDER
 
     assert_equal "hello", result["content"]
   end
 
-  test "partial! + locals via :locals option" do
+  test "partial! + locals without :partial key with :locals key" do
     result = jbuild(<<-JBUILDER)
       json.partial! "partial", locals: { foo: "howdy" }
     JBUILDER
 
     assert_equal "howdy", result["content"]
   end
 
-  test "partial! + locals without :locals key" do
+  test "partial! + locals with :partial key with :locals key" do
     result = jbuild(<<-JBUILDER)
-      json.partial! "partial", foo: "goodbye"
+      json.partial! partial: "partial", locals: { foo: "goodbye" }
+    JBUILDER
+
+    assert_equal "goodbye", result["content"]
+  end
+
+  test "partial! + locals without :partial key without :locals key" do
+    result = jbuild(<<-JBUILDER)
+      json.partial! "partial", foo: "howdy"
+    JBUILDER
+
+    assert_equal "howdy", result["content"]
+  end
+
+  test "partial! + locals with :partial key without :locals key" do
```

Comment:
This is the spec that currently would fail with the above 
```diff
-  "_partial.json.jbuilder"  => "foo ||= 'hello'; json.content foo",
+  "_partial.json.jbuilder"  => "json.content foo",
```
change.